contracts/protocol/modules/LimitOrder.sol

```diff
@@ -143,11 +144,10 @@ abstract contract LimitOrderManager is ClaimableAmountManager {
             if (
                 oldPosition.quantity != 0 &&
                 !oldPosition.quantity.isSameSide(_rawQuantity) &&
+                oldPosition.quantity.abs() <= _quantity &&
```

Comment:
compare new order quantity with old position quantity here, instead of passing to position manager